Std/Data/BitVec/Lemmas.lean

```diff
@@ -155,6 +157,41 @@ theorem truncate_succ (x : BitVec w) :
     have j_lt : j.val < i := Nat.lt_of_le_of_ne (Nat.le_of_succ_le_succ j.isLt) j_eq
     simp [j_eq, j_lt]
 
+/-! ### cast -/
+
+@[simp]
+theorem add_cast {x y : BitVec w} :  (x + y).toNat = (x.toNat + y.toNat) % 2^w := rfl
+@[simp]
+theorem sub_cast {x y : BitVec w} : (x - y).toNat = (x.toNat + (2^w - y.toNat)) % 2^w := rfl
+
+@[simp]
+theorem neg_cast {x : BitVec w} : (-x).toNat = (2^w- x.toNat) % 2^w := by
+  rw [← Nat.zero_add (2^w - x.toNat)]; rfl
+
+@[simp]
+theorem xor_cast {x y: BitVec w} : (x ^^^ y).toNat = x.toNat ^^^ y.toNat:= rfl
+
+/-! ### ring properties -/
+
+theorem eq_sub_of_add_eq {a b c : BitVec w} (h : a + b = c) : a = c - b := by
```

Comment:
```suggestion
theorem eq_sub_of_add_eq (a b c : BitVec w) (h : a + b = c) : a = c - b := by
```
Also move to sub section.

Reply:
This suggestion was a mistake, the original was fine. Sorry. I was probably on auto-pilot :/ This should still be in a section for `sub`.